idb/postgres/postgres_migrations.go

```diff
@@ -4,65 +4,25 @@
 package postgres
 
 import (
-	"bytes"
 	"context"
 	"database/sql"
 	"fmt"
-	"math"
-	"os"
-	"time"
 
-	"github.com/algorand/go-algorand-sdk/crypto"
-	"github.com/algorand/go-algorand-sdk/encoding/msgpack"
 	sdk_types "github.com/algorand/go-algorand-sdk/types"
 
-	"github.com/algorand/indexer/accounting"
-	"github.com/algorand/indexer/api/generated/v2"
 	"github.com/algorand/indexer/idb"
 	"github.com/algorand/indexer/idb/migration"
 	"github.com/algorand/indexer/idb/postgres/internal/encoding"
-	"github.com/algorand/indexer/types"
 )
 
-// rewardsMigrationIndex is the index of m7RewardsAndDatesPart2.
-const rewardsMigrationIndex = 7
+const firstAvailableMigration = 12
 
 func init() {
 	migrations = []migrationStruct{
-		// function, blocking, description
-		{m0fixupTxid, false, "Recompute the txid with corrected algorithm."},
-		{m1fixupBlockTime, true, "Adjust block time to UTC timezone."},
-		{m2apps, true, "Update DB Schema for Algorand application support."},
-		{m3acfgFix, false, "Recompute asset configurations with corrected merge function."},
-
-		// 2.2.2 hotfix
-		{m4accountIndices, true, "Add indices to make sure account lookups remain fast when there are a lot of apps or assets."},
-
-		// Migrations for 2.3.1 release
-		{m5MarkTxnJSONSplit, true, "record round at which txn json recording changes, for future migration to fixup prior records"},
-		{m6RewardsAndDatesPart1, true, "Update DB Schema for cumulative account reward support and creation dates."},
-		{m7RewardsAndDatesPart2, false, "Compute cumulative account rewards for all accounts."},
-
-		// Migrations for 2.3.2 release
-		{m8StaleClosedAccounts, false, "clear some stale data from closed accounts"},
-		{m9TxnJSONEncoding, false, "some txn JSON encodings need app keys base64 encoded"},
-		{m10SpecialAccountCleanup, false, "The initial m7 implementation would miss special accounts."},
-		{m11AssetHoldingFrozen, false, "Fix asset holding freeze states."},
```

Comment:
What do you think about keeping placeholders? There are a few nice benefits:
1) no need to worry about the changing the index?
2) we could replace the existing migration functions with a special `unsupportedMigration` function that errors unconditionally. The comment field could be the last version supported like "2.5.0" to go in a message like "Please downgrade to 2.5.0 and complete migration before installing this version."
3) easier to deprecate future migrations, just delete the function and change the placeholder function to `unsupportedMigration` and note the last supported version.

Reply:
Thanks, I did some something similar.

Reply:
The only issue now is that indexer doesn't exit if can't run a migration, but keeps hanging instead. It might be worth improving the migration engine. What do you think?